torch/csrc/jit/passes/guard_elimination.cpp

```diff
@@ -0,0 +1,147 @@
+#include <torch/csrc/jit/passes/alias_analysis.h>
+#include <torch/csrc/jit/passes/guard_elimination.h>
+#include <memory>
+#include <unordered_set>
+
+namespace torch {
+namespace jit {
+
+struct GuardElimination {
+  GuardElimination(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)),
+        aliasDb_(caffe2::make_unique<AliasDb>(graph_)) {}
+
+  void run() {
+    moveGuardsToDefs(graph_->block());
+    coalesceGuards(graph_->block());
+    eliminateGuards(graph_->block());
+  }
+
+  void moveGuardsToDefs(Block* b) {
+    // alias analysis gets confused if we ask it to move expressions
```

Comment:
This is pretty hacky. Can we just fix alias analysis? It used to be the case that prim::param were not in the list, which probably caused the issue in the first place. That has since been fixed.

Reply:
It might actually just work right now, after reading the code quickly.